scripts/upgrade

```diff
@@ -103,54 +111,54 @@ then
       git fetch
       git checkout stable
       git pull
+      git reset --hard $version_commit
 	popd
 	pushd "$final_path/addon"
       git fetch
       git checkout stable
       git pull
+      git reset --hard $addons_version_commit
 	popd
 # If git is not present upgrade through manual method 
 else
 
     # Create a temporary directory and backup smarty3 folder
-    tmpdir="$(ynh_smart_mktemp 6000)"
-    ynh_script_progression --message="Upgrading source files..."
-    mkdir -p "$tmpdir/view/"
-    cp -a "$final_path/view/smarty3" "$tmpdir/view/smarty3"
+	tmpdir="$(mktemp -d)"
+	ynh_script_progression --message="Upgrading source files..."
+	cp -a "$final_path/view/smarty3" "$tmpdir/view/smarty3"
 
     # Remove the app directory securely
     ynh_secure_remove "$final_path"
 
-    # 1- Clone stable repo 
-    $git clone https://github.com/friendica/friendica.git -b stable "$final_path"
+    # 1 - Clone stable repo 
+    git clone https://github.com/friendica/friendica.git -b stable "$final_path"
+    # Reset branch to the level of update we needed
+    pushd "$final_path"
+    	git reset --hard $version_commit
+    popd
     
     # Remove the addon directory securely
     ynh_secure_remove "$final_path/addon"
```

Comment:
```suggestion
```
No need, since entire `$final_path` was removed earlier:
```
Info: '/var/www/friendica/addon' wasn't deleted because it doesn't exist.
```